pkg/internal/affinityassistant/transformer.go

```diff
@@ -28,32 +28,42 @@ import (
 // NewTransformer returns a pod.Transformer that will pod affinity if needed
 func NewTransformer(_ context.Context, annotations map[string]string) pod.Transformer {
 	return func(p *corev1.Pod) (*corev1.Pod, error) {
-		// Using node affinity on taskRuns sharing PVC workspace, with an Affinity Assistant
-		// is mutually exclusive with other affinity on taskRun pods. If other
-		// affinity is wanted, that should be added on the Affinity Assistant pod unless
-		// assistant is disabled. When Affinity Assistant is disabled, an affinityAssistantName is not set.
+		// Using node affinity on taskRuns sharing PVC workspace.  When Affinity Assistant
+		// is disabled, an affinityAssistantName is not set.
 		if affinityAssistantName := annotations[workspace.AnnotationAffinityAssistantName]; affinityAssistantName != "" {
-			p.Spec.Affinity = nodeAffinityUsingAffinityAssistant(affinityAssistantName)
+			if p.Spec.Affinity == nil {
+				p.Spec.Affinity = &corev1.Affinity{}
+			}
```

Comment:
I think this block is duplicated with the similar block in `mergeAffinityWithAffinityAssistant`-- we don't need both